register.py

```diff
@@ -141,11 +152,12 @@ def request_token(pool, resource):
 
     print("Token request approved!")
 
-    token_dir = htcondor.param["SEC_TOKEN_DIRECTORY"]
+    global TOKEN_DIR
+    if TOKEN_DIR == "":
+        TOKEN_DIR = htcondor.param["SEC_TOKEN_DIRECTORY"]
     token_name = "50-{}-{}-registration".format(alias, resource)
-    token_path = os.path.join(token_dir, token_name)
-
-    logger.debug("Writing token to disk (in {})".format(token_dir))
+    token_path = '/'.join(os.path.join(TOKEN_DIR, token_name).split('\\'))
```

Comment:
```suggestion
    # '/' is an accepted path separator across operating systems
    token_path = os.path.join(TOKEN_DIR, token_name).replace('\\', '/')
```

1.  It's a good idea to explain your reasoning in comments and/or commit messages
2.  Instead of creating a string from an array from a string, let's just create the two strings.